### PR TITLE
Document how web paths are built

### LINKED ISSUE #828: Documenting common pitfalls
Continuing on from https://github.com/liip/LiipImagineBundle/pull/827...

Right now, while not strictly an issue regarding this repository, there's at the very least a common pitfall wherein without a correctly configured Symfony application + reverse proxy, we'll generate links with the wrong scheme (as we utilise Symfony's `Request` class).

I'm not sure if there are any other obvious ones, but I am thinking to remedy this we should start looking at building a FAQ?

### PR DESCRIPTION
**Pull request by greg0ire**
There are common pitfalls that can be easily avoided with some
configuration.
Closes #828

### CODE REVIEW

**Comment by robfrawley** [comment]
@greg0ire @antoligy I reformatted the majority of new information into a `tip` box, and moved the links to the bottoms of the page (our global convention), as well as cleaned up the language a bit. Also, I went ahead and did a local test build of the RST docs to ensure proper formatting. Here's how the new docs look:

liipimaginebundle-rst-docs-pr829

Thoughts on this PR at this point? Looks good to me; will merge shortly pending any comments otherwise.

**Comment by greg0ire** [comment]
Good job! Shall I squash the commits or will you?

File: Resources/doc/cache-resolver/web_path.rst [hunk 1]
```diff
@@ -5,7 +5,21 @@ Web Path Resolver
...
-web path of your application.
+web path of your application. The resulting web path is computed by
+taking a number of factors into account, including the `request context`_
+as provided by the Symfony HTTP kernel.
+
+.. tip::
+
+    The request context is most notably used to determine the HTTP scheme
+    prepended to the final URL. If you use a proxy to offload TLS traffic
+    decryption and need the resolver to generate HTTPS URLS, you will need
```
Comment (greg0ire) [inline Resources/doc/cache-resolver/web_path.rst]: `URLs`, maybe?

Comment (robfrawley) [inline Resources/doc/cache-resolver/web_path.rst]: Heh; already noticed that myself in my follow-up commit. though do please take a look at my final changes and advise if you notice anything else needing cleanup.

**Comment by robfrawley** [comment]
@greg0ire Just pushed an additional commit with some cleanup of the first few paragraphs of the web path resolver docs, as well as some changes to the my initial language with regard to the original intent of this PR. Feel free to squash the commits yourself, but there is no longer a *need* to do so yourself (GitHub PRs now allow for squashing and merging in a single click).

**Comment by robfrawley** [comment]
Also, here is the final rendering with all the prior changes:

liipimaginebundle-rst-docs-pr829-2

**Comment by greg0ire** [comment]
> but there is no longer a need to do so yourself (GitHub PRs now allow for squashing and merging in a single click).

I'm aware of that, it's just to make sure we don't forget it by mistake ;)

File: Resources/doc/cache-resolver/web_path.rst [hunk 2]
```diff
@@ -4,8 +4,26 @@
...
+HTTP kernel.
+
+.. tip::
+
+    The `request context`_ is most notably used to determine the HTTP
+    scheme used for the final URL. If you use a proxy to offload TLS
+    traffic decryption and need the resolver to generate secure URLs,
+    you will need to appropriately configure Symfony's `trusted proxies`_.
+    If you utilize `embedded controllers`_ in your templates, you must
+    add ``localhost`` to your trusted proxy configuration.
```
Comment (greg0ire) [inline Resources/doc/cache-resolver/web_path.rst]: Not so sure about that one… trusted proxies configuration maybe? or "add `localhost` as a trusted proxy"?

Comment (robfrawley) [inline Resources/doc/cache-resolver/web_path.rst]: I prefer "trusted proxies" to rewording the sentence. Pushed a commit reflective of such. Anything else we need to update/change?

**Comment by greg0ire** [comment]
LGTM (PR authors can't approve their own PRs)

**Comment by robfrawley** [comment]
@greg0ire Ignore the base-branch changes; was going to merge immediately into `2.0`, which would not cause the live documentation build process to kick off, and would afford some time for comments prior to merging back to `1.0` (which will immediately cause the build process to kick off).

I'm cautious with documentation changes, so I told my phone to remind me to come back to this PR tomorrow night (I do love Google voice control on Android: "OK Google, remind me [...] tomorrow night"). Assuming no objections or change requests arise, this'll be merged then.

**Comment by greg0ire** [comment]
> I'm cautious with documentation changes, so I told my phone to remind me to come back to this PR tomorrow night (I do love Google voice control on Android: "OK Google, remind me [...] tomorrow night"). 

Jeez, I hope they pay you well 😆

**Comment by antoligy** [comment]
No objections from me, cheers for the contribution and nice work with reformatting!

Off-topic but do we have the documentation standards noted down anywhere so I can try and help catch this stuff in future?

"Ok Google, remind me to pay @robfrawley in :beer:"!

**Comment by robfrawley** [comment]
Thanks @greg0ire!

**Pull request merged**